include/itkCLEImage.h

```diff
@@ -0,0 +1,296 @@
+   * than GPU's. That is because Modified() is called at
+   * the end of each filter in the pipeline so although we
+   * increment GPU's time stamp in GPUGenerateData() the
+   * CPU's time stamp will be increased after that.
+   */
+  void
+  DataHasBeenGenerated() override
+  {
+    Superclass::DataHasBeenGenerated();
+    //if (m_DataManager->IsCPUBufferDirty())
+    {
+      m_DataManager->Modified();
+    }
+  }
+
+protected:
+  void
+  Graft(const DataObject * data) override;
+  CLEImage();
+  ~CLEImage() override;
+  using Superclass::Graft;
+
+private:
+  typename CLEDataManager<TPixel,VImageDimension>::Pointer m_DataManager;
+};
+
+class ITK_TEMPLATE_EXPORT CLEImageFactory : public itk::ObjectFactoryBase
+{
+public:
+  ITK_DISALLOW_COPY_AND_MOVE(CLEImageFactory);
+
+  using Self = CLEImageFactory;
+  using Superclass = itk::ObjectFactoryBase;
+  using Pointer = itk::SmartPointer<Self>;
+  using ConstPointer = itk::SmartPointer<const Self>;
+
+  /** Class methods used to interface with the registered factories. */
+  const char *
+  GetITKSourceVersion() const override
+  {
+    return ITK_SOURCE_VERSION;
+  }
+  const char *
+  GetDescription() const override
+  {
+    return "A Factory for CLEImage";
+  }
+
+  /** Method for class instantiation. */
+  itkFactorylessNewMacro(Self);
+
+  /** Run-time type information (and related methods). */
+  itkTypeMacro(CLEImageFactory, itk::ObjectFactoryBase);
+
+  /** Register one factory of this type  */
+  static void
+  RegisterOneFactory()
+  {
+    auto factory = CLEImageFactory::New();
+
+    itk::ObjectFactoryBase::RegisterFactory(factory);
+  }
+
+private:
+#define OverrideImageTypeMacro(pt, dm)                         \
+  this->RegisterOverride(typeid(itk::Image<pt, dm>).name(),    \
+                         typeid(itk::CLEImage<pt, dm>).name(), \
+                         "CLE Image Override",                 \
+                         true,                                 \
+                         itk::CreateObjectFunction<CLEImage<pt, dm>>::New())
+
+  CLEImageFactory()
+  {
+    /*
+    if (IsGPUAvailable())
+    {
+      // 1/2/3D
+      OverrideImageTypeMacro(unsigned char, 1);
+      OverrideImageTypeMacro(signed char, 1);
+      OverrideImageTypeMacro(int, 1);
+      OverrideImageTypeMacro(unsigned int, 1);
+      OverrideImageTypeMacro(float, 1);
+      OverrideImageTypeMacro(double, 1);
+
+      OverrideImageTypeMacro(unsigned char, 2);
+      OverrideImageTypeMacro(signed char, 2);
+      OverrideImageTypeMacro(int, 2);
+      OverrideImageTypeMacro(unsigned int, 2);
+      OverrideImageTypeMacro(float, 2);
+      OverrideImageTypeMacro(double, 2);
+
+      OverrideImageTypeMacro(unsigned char, 3);
+      OverrideImageTypeMacro(signed char, 3);
+      OverrideImageTypeMacro(int, 3);
+      OverrideImageTypeMacro(unsigned int, 3);
+      OverrideImageTypeMacro(float, 3);
+      OverrideImageTypeMacro(double, 3);
+    }
+    */
+  }
+};
```

Comment:
Check into `itk.Image.{h,hxx}`: see if it does not have any of this content. If so, we may not need it.